omod/src/main/java/org/openmrs/module/messages/resources/ActorResponseRestResource.java

```diff
@@ -0,0 +1,243 @@
+    public Model getCREATEModel(Representation representation) {
+        ModelImpl model = new ModelImpl()
+                .property("patient", new ArrayProperty(new RefProperty("#/definitions/PatientCreate")))
+                .property("person", new ArrayProperty(new RefProperty("#/definitions/PersonCreate")))
+                .property("id", new IntegerProperty())
+                .property("answereTime", new DateProperty())
+                .property("deathDate", new DateProperty())
+                .property("causeOfDeath", new StringProperty())
+                .property("actorResponseType",new ArrayProperty(new RefProperty("#/definitions/ActorResponseTypeAttributeCreate")))
+                .property("question",new ArrayProperty(new RefProperty("#/definitions/ActorResponseTypeQuestionCreate")))
+                .property("attributes", new ArrayProperty(new RefProperty("#/definitions/PersonAttributeCreate")));
+
+        model.setRequired(Arrays.asList("question", "response"));
+        return model;
+    }
+    
+    @Override
+    public Model getUPDATEModel(Representation representation) {
+        ModelImpl model = new ModelImpl()
+                .property("patient", new ArrayProperty(new RefProperty("#/definitions/PatientCreate")))
+                .property("person", new ArrayProperty(new RefProperty("#/definitions/PersonCreate")))
+                .property("sourceId", new StringProperty())
+                .property("textResponse", new StringProperty())
+                .property("id", new IntegerProperty())
+                .property("answereTime", new StringProperty())
+                .property("deathDate", new DateProperty())
+                .property("causeOfDeath", new StringProperty())
+                .property("actorResponseType",
+                        new ArrayProperty(new RefProperty("#/definitions/ActorResponseTypeAttributeCreate")))
+                .property("question",
+                        new ArrayProperty(new RefProperty("#/definitions/ActorResponseTypeQuestionCreate")))
+                .property("attributes", new ArrayProperty(new RefProperty("#/definitions/PersonAttributeCreate")));
+
+        model.setRequired(Arrays.asList("question", "response"));
+        return model;
+    }
+    
+    /**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getPropertiesToExposeAsSubResources()
+	 */
+	@Override
+    public List<String> getPropertiesToExposeAsSubResources() {
+        return Arrays.asList("actorResponseType", "question", "attributes");
+    }
+ 
+    @Override
+    public ActorResponse newDelegate() {
+        return new ActorResponse();
+    }
+
+    @Override
+    public ActorResponse save(ActorResponse delegate ) {
+        return Context.getService(ActorResponseService.class).saveOrUpdate(delegate);
+
+    }
+
+    @Override
+    protected void delete(ActorResponse delegate, String textQuestion, RequestContext textResponse ) throws ResponseException {
```

Comment:
why textQuestion and textResponse params are here? They are not used

Reply:
Resolved thanks

Reply:
You changed only param names. You still have two unsed params - they are required because you override delete method from interface, right?
Second remark - you have condition "if response != null" then "do nothing". It does it make sense. In this case you always delete null entity